ballerina-tests/tests/38_interceptors.bal

```diff
@@ -427,3 +428,206 @@ isolated function testInterceptorsWithTables() returns error? {
+            }
+        }
+    };
+    check validateWebSocketResponse(wsClient, expectedPayload);
+}
+
+@test:Config {
+    groups: ["interceptors", "subscriptions"]
+}
+isolated function testInterceptorsReturnBeforeResolverWithSubscription() returns error? {
+    string document = string `subscription { messages }`;
+    string url = "ws://localhost:9099/subscription_interceptor5";
+    websocket:Client wsClient1 = check new(url);
+    websocket:Client wsClient2 = check new(url);
+    check writeWebSocketTextMessage(document, wsClient1);
+    check writeWebSocketTextMessage(document, wsClient2);
+    json expectedPayload = {data: {messages: 1}};
+    foreach int i in 1 ..< 4 {
+        check validateWebSocketResponse(wsClient1, expectedPayload);
+        check validateWebSocketResponse(wsClient2, expectedPayload);
+    }
+}
+
+@test:Config {
+    groups: ["interceptors", "subscriptions"]
+}
+isolated function testInterceptorsDestructiveModificationWithSubscription() returns error? {
+    string document = string `subscription { messages }`;
+    string url = "ws://localhost:9099/subscription_interceptor6";
+    websocket:Client wsClient1 = check new(url);
+    websocket:Client wsClient2 = check new(url);
+    check writeWebSocketTextMessage(document, wsClient1);
+    check writeWebSocketTextMessage(document, wsClient2);
+    json expectedPayload = {
+        errors:[
+            {
+                message: "Invalid return type in Interceptor \"DestructiveModification\". Expected type Int!",
+                locations: [
+                    {
+                        line:1,
+                        column:16
+                    }
+                ],
+                path: ["messages"]
+            }
+        ],
+        data:null
+    };
+    foreach int i in 1 ..< 4 {
+        check validateWebSocketResponse(wsClient1, expectedPayload);
+        check validateWebSocketResponse(wsClient2, expectedPayload);
+    }
+}
+
+@test:Config {
+    groups: ["interceptors", "subscriptions"]
+}
+isolated function testInterceptorsWithSubscribersRunSimultaniously1() returns error? {
+    final string document = string `subscription { messages }`;
+    string url = "ws://localhost:9099/subscription_interceptor1";
+    final websocket:Client wsClient1 = check new(url);
+    final websocket:Client wsClient2 = check new(url);
+    final websocket:Client wsClient3 = check new(url);
+    worker A returns error? {
+        check writeWebSocketTextMessage(document, wsClient1);
+        foreach int i in 1 ..< 4 {
+            json expectedPayload = {data: {messages: i * 5 - 5}};
+            check validateWebSocketResponse(wsClient1, expectedPayload);
+        }
+    }
+    worker B returns error? {
+        check writeWebSocketTextMessage(document, wsClient2);
+        foreach int i in 1 ..< 4 {
+            json expectedPayload = {data: {messages: i * 5 - 5}};
+            check validateWebSocketResponse(wsClient2, expectedPayload);
+        }
+    }
```

Comment:
Shouldn't we wait for these workers to be finished?

Reply:
Missed that. Fixed via https://github.com/ballerina-platform/module-ballerina-graphql/pull/939/commits/7f0b4abf6f5842435c1015c7791661b11b0bd21f